nuxeo-wopi/src/main/java/org/nuxeo/wopi/jaxrs/FilesEndpoint.java

```diff
@@ -678,4 +796,53 @@ protected String getDocumentURL(DocumentModel doc) {
         return null;
     }
 
+    protected void logRequest(String operation, String... headers) {
+        log.debug("Request: repository={} docId={} xpath={} user={} fileId={} operation={}{}", doc::getRepositoryName,
+                doc::getId, () -> xpath, session::getPrincipal, () -> fileId, () -> operation,
+                () -> getHeaderString(headers));
+    }
+
+    protected void logCondition(String condition) {
+        logCondition(() -> condition);
+    }
+
+    protected void logCondition(Supplier<String> condition) {
+        log.debug("Condition: repository={} docId={} xpath={} user={} fileId={} {}", doc::getRepositoryName, doc::getId,
+                () -> xpath, session::getPrincipal, () -> fileId, condition::get);
```

Comment:
You need to give `condition::get` instead of `condition` directly cause of type ?

Reply:
Yes, doesn't compile otherwise: `The target type of this expression must be a functional interface`

Reply:
Why not having just
```
protected void logCondition(String condition) {
        log.debug("Condition: repository={} docId={} xpath={} user={} fileId={} {}", doc::getRepositoryName, doc::getId,
                () -> xpath, session::getPrincipal, () -> fileId, () -> condition);
}
```

Reply:
Because I want the condition to not be evaluated unless DEBUG logging is activated, such as in https://github.com/nuxeo/nuxeo-wopi/blob/task-NXP-25583-add-debug-logs/nuxeo-wopi/src/main/java/org/nuxeo/wopi/jaxrs/FilesEndpoint.java#L297